worldedit-core/src/main/java/com/fastasyncworldedit/core/util/MultiFuture.java

```diff
@@ -0,0 +1,55 @@
+package com.fastasyncworldedit.core.util;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class MultiFuture implements Future<Object[]> {
```

Comment:
Is there anything that can't be covered by using `CompletableFuture#allOf(...)`? The current implementation of `get` with timeout here is insufficient in cases where e.g. the first future is the only one that isn't done but requires more than `timeout  / size` to complete.

Reply:
You can't make a completable future from futures. I did look at using it, but using completeable future isn't ideal